TELCODOCS-1067: Create KMMO doc: Firmware support (#58256)
openshift/openshift-docs: modules/kmm-configuring-the-lookup-path-on-nodes.adoc

// Module included in the following assemblies:
//
// * hardware_enablement/kmm-kernel-module-management.adoc

:_content-type: PROCEDURE
[id="kmm-configuring-the-lookup-path-on-nodes_{context}"]
= Configuring the lookup path on nodes

On OpenShift nodes, the set of default lookup paths for firmwares does not include `/var/lib/firmware`.

.Procedure

. Use the Machine Config Operator to create a `MachineConfig` resource that contains the `/var/lib/firmware` path:
+
[source,yaml]
----
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:

>     machineconfiguration.openshift.io/role: worker

Reviewer: Maybe add a note that the label is user configurable based on his needs . In case of SNOs it should be `control-plane` or `master`

Author: /Added a callout: **<1> You can configure the label based on your needs. In the case of SNOs, use either `control-pane` or `master`.**

  name: 99-worker-kernel-args-firmware-path
spec:
  kernelArguments:
    - 'firmware_class.path=/var/lib/firmware'
----

> . Reboot all of the worker nodes.

Reviewer: By applying the MachineConfig, the nodes will get automatically rebooted

Author: Added this to the step.

// Module included in the following assemblies:
//
// * hardware_enablement/kmm-kernel-module-management.adoc

:_content-type: PROCEDURE
[id="kmm-configuring-the-lookup-path-on-nodes_{context}"]
= Configuring the lookup path on nodes

> On OpenShift nodes, the set of default lookup paths for firmwares does not include `/var/lib/firmware`.

Reviewer 2: ```suggestion
On {product-title} nodes, the set of default lookup paths for firmwares does not include the `/var/lib/firmware` path.
```

Author: Made these changes.

.Procedure

> . Use the Machine Config Operator to create a `MachineConfig` resource that contains the `/var/lib/firmware` path:

Author: changed.

+
[source,yaml]
----
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: worker <1>
  name: 99-worker-kernel-args-firmware-path
spec:
  kernelArguments:
    - 'firmware_class.path=/var/lib/firmware'
----

> <1> You can configure the label based on your needs. In the case of SNOs, use either `control-pane` or `master`.

Author: changed.

> <1> You can configure the label based on your needs. In the case of SNOs, use either `control-pane` or `master` objects.

Reviewer 3: See OpenShift glossary of [single-node OpenShift](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#single-node-openshift). You can use the `{sno}` attribute.
)
```suggestion
<1> You can configure the label based on your needs. In the case of {sno}, use either `control-pane` or `master` objects.
```

Author: changed.

> . By applying the MachineConfig, the nodes are automatically rebooted.

Reviewer 2: I believe the `MachineConfig` here should be in backtics
```suggestion
. By applying the `MachineConfig` CR, the nodes are automatically rebooted.
```

Author: changed.

> . Reboot all of the worker nodes. By applying the MachineConfig, the nodes are automatically rebooted.

Reviewer: Please remove the _Reboot all of the worker nodes._ part and leave only the _By applying..._

Author: @cdvultur Change made. Please /LGTM comment.
